src/main/java/ui/windows/StatsDelegator.java

```diff
@@ -26,6 +31,11 @@ public StatsDelegator(StatsController controller) {
     public void actionPerformed(ActionEvent e) {
         JTextField inputNameField = (JTextField)e.getSource();
         String username = inputNameField.getText();
-        controller.tryRequestUser(username);
+        try {
+            PlayerInformation player = database.loadPlayer(username);
```

Comment:
loading players should be done in the StatsController class to adhere to SRP and to not skip layers

delegators should only handle button presses and tell the controller what to do